src/InfoViz/Native/FieldSelector/index.js

```diff
@@ -71,8 +72,8 @@ function fieldSelector(publicAPI, model) {
     function renderField(d, i) {
       const field = model.provider.getField(d);
       const fieldContainer = d3.select(this);
-      let legendCell = fieldContainer.select('.legend');
-      let fieldCell = fieldContainer.select('.field');
+      let legendCell = fieldContainer.select(`.${style.legend}`);
+      let fieldCell = fieldContainer.select(`.${style.fieldName}`);
```

Comment:
Hi Seb,
I added a revision of HistogramSelector that shows how I propose to use
empty CSS Module styles for selection.

Here's why - it avoids an external style sheet adding styles for one of the
classes that we intend to use for selection only. Since we've already had
name conflicts with the ERDC ERS portal, it seems prudent to make sure our
selection styles are private. What do you think of this proposal?

I neglected to change FieldSelector to match HistogramSelector. I can do
that today.

Thanks,
Aron

On Wed, Jun 22, 2016 at 11:53 PM, Sebastien Jourdain <
notifications@github.com> wrote:

> In src/InfoViz/Native/FieldSelector/index.js
> <https://github.com/Kitware/paraviewweb/pull/139#discussion_r68172349>:
>
> > @@ -71,8 +72,8 @@ function fieldSelector(publicAPI, model) {
> >      function renderField(d, i) {
> >        const field = model.provider.getField(d);
> >        const fieldContainer = d3.select(this);
> > -      let legendCell = fieldContainer.select('.legend');
> > -      let fieldCell = fieldContainer.select('.field');
> > +      let legendCell = fieldContainer.select(`.${style.legend}`);
> > +      let fieldCell = fieldContainer.select(`.${style.fieldName}`);
>
> You should not use style for making JavaScript selection. Instead use a
> class like jsLegendContainer.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Kitware/paraviewweb/pull/139/files/49e752b4bd5d24969b09d143e5e5e41c52559c2a#r68172349>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AS_pna3EdXciePtmmNvldlW6vEfvfUEbks5qOgM5gaJpZM4I7Jva>
> .
>